service/github/github.go

```diff
@@ -321,18 +321,39 @@ func (g *PullRequest) diffUsingGitCommand(ctx context.Context) ([]byte, error) {
 		return nil, err
 	}
 
-	headSha := pr.GetHead().GetSHA()
+	head := pr.GetHead()
+	headSha := head.GetSHA()
 
-	mergeBaseBytes, err := exec.Command("git", "merge-base", headSha, pr.GetBase().GetSHA()).Output()
+	commitsComparison, _, err := g.cli.Repositories.CompareCommits(ctx, g.owner, g.repo, headSha, pr.GetBase().GetSHA(), nil)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get merge-base commit: %w", err)
+		return nil, err
 	}
 
-	mergeBase := strings.Trim(string(mergeBaseBytes), "\n")
+	mergeBaseSha := commitsComparison.GetMergeBaseCommit().GetSHA()
+
+	if os.Getenv("REVIEWDOG_SKIP_GIT_FETCH") != "true" {
+		remoteRepo := "reviewdog_origin"
+
+		if _, err := exec.Command("git", "remote", "get-url", remoteRepo).CombinedOutput(); err != nil {
+			log.Printf("failed to run git remote get-url: %s", err)
+
+			_, err := exec.Command("git", "remote", "add", remoteRepo, head.GetRepo().GetHTMLURL()).CombinedOutput()
+			if err != nil {
+				return nil, fmt.Errorf("failed to run git remote add: %w", err)
+			}
+		}
+
+		for _, sha := range []string{mergeBaseSha, headSha} {
```

Comment:
Why do we need to fetch `headSha`? 
It didn't work without fetching it?

Reply:
https://github.com/reviewdog/reviewdog/actions/runs/8746978073/job/24004712656?pr=1717  ( https://github.com/reviewdog/reviewdog/pull/1717/commits/a5a03bda8166923b674c24eab300385361b41225 )

```
reviewdog: fail to get diff: failed to run git diff: exit status 128
```

It look like the above if I don't fetch the head sha.
This is probably because `actions/checkout` gets the merged commit of head and base branch by default.

https://github.com/reviewdog/reviewdog/actions/runs/8746978073/job/24004712656?pr=1717

```
/usr/bin/git checkout --progress --force refs/remotes/pull/1717/merge
...
  HEAD is now at f62528f Merge a5a03bda8166923b674c24eab300385361b41225 into e34989bf40a5fa7aa4cf5d9f45f3f2e5d1edd3c9
/usr/bin/git log -1 --format='%H'
'f62528f5a35c3964b6441798dae6f9081d8773d7'
```


In the above example, https://github.com/reviewdog/reviewdog/pull/1717/commits/a5a03bda8166923b674c24eab300385361b41225 is head sha, https://github.com/reviewdog/reviewdog/commit/e34989bf40a5fa7aa4cf5d9f45f3f2e5d1edd3c9 is base sha and https://github.com/reviewdog/reviewdog/commit/f62528f5a35c3964b6441798dae6f9081d8773d7 is merged commit sha.

Reply:
It might be a good idea to get the head sha using `/usr/bin/git log -1 --format='%H'` instead of GitHub API.

Reply:
Ok, thank you for the investigation and explanation!
I'm fine with leaving it as is.